activerecord/lib/active_record/connection_adapters/trilogy_adapter.rb

```diff
@@ -205,10 +202,15 @@ def translate_exception(exception, message:, sql:, binds:)
           end
 
           case exception
-          when Errno::EPIPE, SocketError, IOError
+          when ->(e) { e.is_a?(::Trilogy::TimeoutError) }
```

Comment:
Why not just ?

```suggestion
          when ::Trilogy::TimeoutError
```

Reply:
> Edit 2: Oops, that doesn't work either because I didn't realize how SystemCallError#=== works.

That ends up matching all `ETIMEDOUT` because the errno is the same

Reply:
Argh, sorry I missed that part.

Reply:
```rb
Trilogy::TimeoutError === Errno::ETIMEDOUT.new
#=> true
```